tests/components/lunatone/test_light.py

```diff
@@ -0,0 +1,142 @@
+"""Tests for the Lunatone integration."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from lunatone_rest_api_client import Device
+from lunatone_rest_api_client.models import ControlData
+import pytest
+
+from homeassistant.components.lunatone.const import DOMAIN
+from homeassistant.components.lunatone.light import LunatoneLight
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from . import DEVICE_DATA_LIST, SERIAL_NUMBER, setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_lunatone_device(mock_lunatone_auth: AsyncMock) -> Device:
+    """Mock a Lunatone device object."""
+    device = Device(mock_lunatone_auth, DEVICE_DATA_LIST[0])
+    device.async_update = AsyncMock()
+    device.async_control = AsyncMock()
+    return device
+
+
+@pytest.fixture
+def mock_lunatone_devices_coordinator(
+    mock_lunatone_devices: AsyncMock,
+) -> Generator[AsyncMock]:
+    """Mock a Lunatone devices coordinator object."""
+    with (
+        patch(
+            "homeassistant.components.lunatone.LunatoneDevicesDataUpdateCoordinator",
+            autospec=True,
+        ) as mock_coordinator,
+        patch(
+            "homeassistant.components.lunatone.light.LunatoneDevicesDataUpdateCoordinator",
+            new=mock_coordinator,
+        ),
+    ):
+        coordinator = mock_coordinator.return_value
+        coordinator.devices_api = mock_lunatone_devices
+        coordinator.device_api_mapping = {
+            device.id: device for device in mock_lunatone_devices.devices
+        }
+        coordinator.last_update_success = True
+        yield coordinator
+
+
+@pytest.fixture
+def light_entity(
+    mock_lunatone_devices_coordinator: AsyncMock, mock_lunatone_device: Device
+) -> LunatoneLight:
+    """Create a LunatoneLight entity using the mock device."""
+
+    async def fake_control(control_data: ControlData):
+        if control_data.switchable is not None:
+            mock_lunatone_device._data.features.switchable.status = (
+                control_data.switchable
+            )
+
+    mock_lunatone_device.async_control.side_effect = fake_control
+
+    entity = LunatoneLight(
+        mock_lunatone_devices_coordinator, mock_lunatone_device.id, SERIAL_NUMBER
+    )
+    entity._device = mock_lunatone_device
+    entity.async_write_ha_state = MagicMock()
+    return entity
+
+
+async def test_setup(
+    hass: HomeAssistant,
+    mock_lunatone_devices: AsyncMock,
+    mock_lunatone_info: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test the Lunatone configuration entry loading/unloading."""
+    await setup_integration(hass, mock_config_entry)
+
+    for device_data in DEVICE_DATA_LIST:
```

Comment:
not strictly necessary, but you could use snapshots for this if you want, so that the full entity data and state gets validated